pkg/utils/utils.go

```diff
@@ -67,3 +72,41 @@ func ContainsAny[T comparable](s1 []T, s2 []T) bool {
 	}
 	return false
 }
+
+// CompareVersionStrings compares two version strings numerically
+// Returns: -1 if v1 < v2, 0 if v1 == v2, 1 if v1 > v2
+func CompareVersionStrings(v1, v2 string) int {
```

Comment:
Rpm version comparison is kinda insane (https://blog.jasonantman.com/2014/07/how-yum-and-rpm-compare-versions/), and while i think this likely would work for this redhat-release version just fine,  i worry someone might use it in the future for other things.  What do you think about using https://github.com/knqyf263/go-rpm-version to do the comparison, it uses a proper rpmversion comparison https://github.com/knqyf263/go-rpm-version/blob/master/version.go#L118

Reply:
The version we use is already separated from the epoch and release, so it'd be harder to misuse. But this function simplifies things anyway so I'll go ahead and switch to it.